### PR TITLE
add solution for the index

### PR DESCRIPTION
**Pull request by wolfrednicolas**

### CODE REVIEW

**Review by drewroberts — APPROVED** [review]
Thank you! 🏇

File: database/migrations/2020_01_01_160000_create_search_volumes_table.php [hunk 1]
```diff
@@ -10,6 +10,8 @@ class CreateSearchVolumesTable extends Migration
 {
     public function up()
     {
+        Schema::defaultStringLength(191);
```
Comment (jasonmccreary) [inline database/migrations/2020_01_01_160000_create_search_volumes_table.php:13]: From memory, this should only be necessary for MariaDB or MySQL less than 5.7.7. I'm not sure how old that version is now or if you'll want to support it.

Furthermore, most [docs](https://laravel.com/docs/master/migrations#index-lengths-mysql-mariadb) recommend adding this in your `AppServiceProvider`, not packages. So it probably shouldn't go here.

**Review by jasonmccreary — COMMENTED** [review]
I would revert this...

**Comment by wolfrednicolas** [comment]
> I would revert this...

Hi @jasonmccreary , If you revert it. make sure the migration is working properly, because of this `$table->unique(['engine', 'provider', 'keyword_id', 'range', 'range_value'], 'search_volume_unique');`

**Comment by jasonmccreary** [comment]
It will, provided the application is running in an environment which meets modern requirements or the parent application properly sets this.

My suggestion to revert is not that it is "not needed", more that it should not be done in this package.

What environment were you running this in that cased the issue?

**Comment by pdbreen** [comment]
The problem is the index includes 4 default length strings which exceeds MySql index length.  The proper fix should be to establish appropriate max lengths for those 4 string fields ['engine', 'provider', 'range', 'range_value'] instead of using the default length.  Alternatively, move those string fields into related tables and replace with their corresponding IDs in this table/this index.

**Comment by jasonmccreary** [comment]
Sure, if the index may be changed, that would be ideal for multiple reasons. As is though, I don't think the `defaultStringLength` _belongs here_.

**Comment by pdbreen** [comment]
I'm in full agreement.  This change should be reverted.  I was just passing on what should be done after reverting it to fix the issue properly.

**Comment by drewroberts** [comment]
Thank you all. I'm going to revert it. @pdbreen Could you fix the issue properly? I couldn't figure it out the other day.

**Comment by pdbreen** [comment]
#93

**Pull request merged**